source/reading.Rmd

```diff
@@ -1159,124 +1135,176 @@ is that data owners have much more control over the data they provide to users.
+nasa_data <- resp_body_json(response)
+nasa_data
 ```
 
-When you call the `get_timelines` for the first time (or any other `rtweet` function that accesses the API), 
-you will see a browser pop-up that looks something like Figure \@ref(fig:01-tidyverse-authorize).
+```{r hidden_query, echo = FALSE, warning = FALSE, message = FALSE}
+library(jsonlite)
 
-(ref:01-tidyverse-authorize) The `rtweet` authorization prompt.
+nasa_data <- read_json("data/nasa.json")
 
-```{r 01-tidyverse-authorize, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = "(ref:01-tidyverse-authorize)", fig.retina = 2, out.width="100%"}
-knitr::include_graphics("img/reading/authorize_question.png")
+# the last entry in the stored data is July 13, 2023, so print that
+nasa_data[[74]] 
 ```
 
-This is the `rtweet` package asking you to provide your own Twitter account's login information.
-When `rtweet` talks to the Twitter API, it uses your account information to authenticate requests;
-Twitter then can keep track of how much data you're asking for, and how frequently you're asking.
-If you want to follow along with this example using your own Twitter account, you should read
-over the list of permissions you are granting `rtweet` *very carefully* and make sure you are comfortable
-with it. Note that `rtweet` can be used to manage most aspects of your account (make posts, follow others, etc.),
-which is why `rtweet` asks for such extensive permissions.
-If you decide to allow `rtweet` to talk to the Twitter API using your account information, then 
-input your username and password and hit "Sign In." Twitter will probably send you an email to say
-that there was an unusual login attempt on your account, and in that case you will have to take
-the one-time code they send you and provide that to the `rtweet` login page too. 
-
-> **Note:** Every API has its own way to authenticate users when they try to access data. Many APIs require you to
-> sign up to receive a *token*, which is a secret password that you input into the R package (like `rtweet`) 
-> that you are using to access the API. 
-
-With the authentication setup out of the way, let's run the `get_timelines` function again to actually access
-the API and take a look at what was returned:
+We can obtain more records at once by using the `start_date` and `end_date` parameters, as
+shown in the table of parameters in \@ref(fig:NASA-API-parameters).
+Let's obtain all the records between May 1, 2023, and July 13, 2023, and store the result
+in an object called `nasa_data`; now the response
+will take the form of an R *list* (you'll learn more about these in Chapter \@ref(wrangling)),
+with one item similar to the above for each of the 74 days between the start and end dates:
 
 ```r
-tidyverse_tweets <- get_timelines('tidyverse', n = 400)
-tidyverse_tweets
+req <- request("https://api.nasa.gov/planetary/apod?api_key=YOUR_API_KEY&start_date=2023-05-01&end_date=2023-07-13")
+response <- req_perform(req)
```

Comment:
Perhaps we should give these variables different names from the ones above to avoid confusion. At least `nasa_data`